iroha_p2p/src/network.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+use std::fmt::{Debug, Formatter};
+use std::io::{Error, ErrorKind};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use async_stream::stream;
+use futures::Stream;
+use rand::{rngs::StdRng, SeedableRng, seq::SliceRandom};
+use tokio::io;
+use tokio::io::{AsyncReadExt, AsyncWriteExt, Interest};
+use tokio::net::{TcpListener, TcpStream};
+use tokio::sync::{Mutex, MutexGuard, RwLock};
+use tokio::time::Duration;
+
+use iroha_actor::{Actor, Addr, Context, ContextHandler, Handler, MessageResponse, AlwaysAddr};
+use iroha_actor::broker::Broker;
+use iroha_crypto::{Hash, KeyPair, PublicKey};
+use iroha_error::Result;
+use iroha_logger::{debug, error, info, trace, warn};
+
+use crate::message::Message;
+use crate::peer::{Peer, State};
+use iroha_data_model::peer::prelude::PeerId;
+
+pub struct Network {
+    peers: HashMap<PeerId, Addr<Peer>>,
+    listener: Option<TcpListener>,
+    addr: Option<Addr<Self>>,
```

Comment:
You don't need to have and address of yourself here. It is always kept in context of actor, so you could just ask it for it.

Reply:
I use this address in `connect_peer()` that will be called from outside, maybe from Torii, don't know for now.